tiledb/sm/group/group_directory.cc

```diff
@@ -66,7 +73,7 @@ GroupDirectory::GroupDirectory(
     , loaded_(false) {
   auto st = load();
```

Comment:
This PR doesn't need to be expanded, so don't take the following comment as grounds for expanding scope unless you _really_ want to.

`load()` is basically an `init()` function. It's only called once (and that's here) and it could be inlined entirely. And there are other functions only called once and not independently unit tested. But also the way it's written if the URI isn't actually a valid group it's going to perform non-optimally because of how it's written. (Specifically, it must wait for all I/O to complete before it's able to signal an error.) So don't touch it.

The one change that might have been in scope here would be to de-`Status` the function `load()`. But whatever. That change will be moot with a rework.